src/ore/src/ssh_key.rs

```diff
@@ -0,0 +1,123 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License in the LICENSE file at the
+// root of this repository, or online at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Utilities for generating and managing SSH keys.
+
+use openssl::pkey::{PKey, Private};
+use serde::{Deserialize, Serialize};
+use ssh_key::private::{Ed25519Keypair, Ed25519PrivateKey, KeypairData};
+use ssh_key::public::Ed25519PublicKey;
+use ssh_key::{LineEnding, PrivateKey};
+
+/// Keeps track of a public-private keypair, stored in the OpenSSH format
+#[derive(Debug, Eq, PartialEq, PartialOrd, Ord, Clone, Serialize, Deserialize)]
+pub struct SshKeypair {
+    public_key: Vec<u8>,
+    private_key: Vec<u8>,
+}
+
+impl SshKeypair {
+    /// Generate a new keypair for SSH connections.
+    ///
+    /// Ed25519 keys are generated via OpenSSL, using [`ssh_key`] to convert
+    /// them into the proprietary OpenSSH format.
+    pub fn new() -> Result<SshKeypair, anyhow::Error> {
+        let openssl_key = PKey::<Private>::generate_ed25519()?;
```

Comment:
we are generating our own ssh keys for each connection? is that preferred over just using whatever keys we have in the container?

Reply:
Yes! The public part of each key is going to be exposed to users to let them give us access to their bastion host to reach their Postgres db. The idea is that even though we are generating the keys, the customers manage their lifecycle: they can rotate them at will (via `ALTER CONNECTION conn_name ROTATE KEYS`).